voxblox/src/simulation/simulation_world.cc

```diff
@@ -95,4 +95,69 @@ void SimulationWorld::getPointcloudFromViewpoint(
   }
 }
 
+void SimulationWorld::getNoisyPointcloudFromViewpoint(
+    const Point& view_origin, const Point& view_direction,
+    const Eigen::Vector2i& camera_res, FloatingPoint fov_h_rad,
+    FloatingPoint max_dist, FloatingPoint noise_sigma, Pointcloud* ptcloud,
+    Colors* colors) {
+  // Focal length based on fov.
+  FloatingPoint focal_length = camera_res.x() / (2 * tan(fov_h_rad / 2.0));
+
+  // Calculate transformation between nominal camera view direction and our
+  // view direction. Nominal view is positive x direction.
+  Point nominal_view_direction(1.0, 0.0, 0.0);
+  Eigen::Quaternion<FloatingPoint> ray_rotation =
+      Eigen::Quaternion<FloatingPoint>::FromTwoVectors(nominal_view_direction,
+                                                       view_direction);
+
+  // Now actually iterate over all pixels.
+  for (int u = -camera_res.x() / 2; u < camera_res.x() / 2; ++u) {
+    for (int v = -camera_res.y() / 2; v < camera_res.y() / 2; ++v) {
+      Point ray_camera_direction =
+          Point(1.0, u / focal_length, v / focal_length);
+      Point ray_direction = ray_rotation * (ray_camera_direction).normalized();
+
+      // Cast this ray into every object.
+      bool ray_valid = false;
+      FloatingPoint ray_dist = max_dist;
+      Point ray_intersect = Point::Zero();
+      Color ray_color;
+      for (const std::unique_ptr<Object>& object : objects_) {
+        Point object_intersect;
+        FloatingPoint object_dist;
+        bool intersects =
+            object->getRayIntersection(view_origin, ray_direction, max_dist,
+                                       &object_intersect, &object_dist);
+        if (intersects) {
+          if (!ray_valid || object_dist < ray_dist) {
+            ray_valid = true;
+            ray_dist = object_dist;
+            ray_intersect = object_intersect;
+            ray_color = object->getColor();
+          }
+        }
+      }
+      if (ray_valid) {
+        // Apply noise now!
+        FloatingPoint noise = getNoise(noise_sigma);
+        ray_dist += noise;
```

Comment:
no noise on the bearing?

Reply:
Decided to just do a simplified model of noise (so yes, only in z). I'll add a comment. But at any rate, I'm not sure how to properly do noise in the bearing -- perturb the vector that I put into the lookup? Since the output bearings are all fixed (i.e., it's the inverse projection onto the image plane).